ambry-router/src/main/java/com.github.ambry.router/PutOperation.java

```diff
@@ -324,6 +336,38 @@ void fillChunks() {
   }
 
   /**
+   * Called whenever the channel has data but no free or building chunk is available to be filled.
+   */
+  private void maybeStartTrackingWaitTime() {
+    if (startTimeForChunkAvailabilityWaitMs == 0) {
+      // this is the first point in time after the last chunk filling (if any) when the filling was blocked due to
+      // chunk unavailability, so mark this time.
+      startTimeForChunkAvailabilityWaitMs = time.milliseconds();
+    } else {
+      // the wait was already initiated, so do nothing.
+    }
+  }
+
+  /**
+   * Called whenever the channel has data and there is a free or building chunk available to be filled.
+   */
+  private void maybeUpdateTrackedWaitTime() {
+    if (startTimeForChunkAvailabilityWaitMs != 0) {
+      // this is the first point in time since the last wait that a chunk became available for filling.
+      waitTimeForCurrentChunkAvailabilityMs = time.milliseconds() - startTimeForChunkAvailabilityWaitMs;
+      startTimeForChunkAvailabilityWaitMs = 0;
+    }
+  }
+
+  /**
+   * Reset time variables associated with tracking wait times.
+   */
+  private void resetWaitTimeTracking() {
```

Comment:
don't think this function is required since the only important statement here is `waitTimeForCurrentChunkAvailabilityMs = 0;` right? The other should be 0 already.